docs/src/transExt.md

```diff
@@ -0,0 +1,94 @@
+```@meta
+CurrentModule = Singular
+```
+
+# Function fields
+
+Function fields are implemented via the Singular `n_transExt` type for prime fields of any characteristic.
+
+The associated function field is represented by a parent object which can be constructed
+by a call to the `FunctionField` constructor.
+
+The types of the parent objects and elements of the associated function fields are given
+in the following table according to the library providing them.
+
+ Library        | Element type  | Parent type
+----------------|---------------|--------------------
+Singular        | `n_transExt`        | `Singular.N_FField`
+
+All function field element types belong directly to the abstract type `FieldElem` and
+all the parent object types belong to the abstract type `Field`.
+
+## Function field functionality
+
+Singular.jl function fields implement the Field interface of AbstractAlgebra.jl.
+
+[https://nemocas.github.io/AbstractAlgebra.jl/fields.html](https://nemocas.github.io/AbstractAlgebra.jl/fields.html)
+
+Below, we describe the functionality that is specific to Singular function field and not
+already listed at the given link.
+
+### Constructors
+
+The following constructors are available to create function fields and their elements.
+
+```@docs
+Singular.FunctionField(::Field, ::Array{String, 1}; ::Bool)
```

Comment:
An examples section would be useful here so people can look and easy see how to use this constructor.

Reply:
DOne